src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Encrypt.cs

```diff
@@ -182,7 +182,6 @@ internal sealed partial class ManagedPkcsPal : PkcsPal
 
                 if (contentInfo.ContentType.Value == Oids.Pkcs7Data)
                 {
-                    toEncrypt = EncodeOctetString(toEncrypt);
```

Comment:
I feel like a test for this is possible.  A 31-byte input as `id-data` should turn into 32-bytes after CBC blocking.  Under the previous scheme it would have turned into a 33-byte plaintext, therefore spills into 48 (DES/3DES) or 64 (AES) bytes of ciphertext.

I can't think of anywhere in the enveloping process that has non-deterministic output length; so the same inputs at 31 bytes of plaintext should always produce the same output size (now in agreement on Windows and Linux), and increasing the plaintext to 32 bytes should make it symmetric-block-size bigger (the easiest form of the test).

Reply:
That's clever - I'll incorporate that idea.

Reply:
@bartonjs Sorry for the hold up - I added a test.

>A 31-byte input as id-data should turn into 32-bytes

Not sure if you were going for 31 specifically, I wrote it as 15 / 16 since AES has a fixed block size of 16 bytes.